### DISCUSSION TITLE
Error in site replication while adding distributed pool to a single node with site replication

### DISCUSSION THREAD

**Discussion by faryar76** [minio/minio#17150] [Q&A] [answered]
I am trying to connect a distributed pool to a single node with site replication. However, I am encountering an error with the signature calculation while replicating the site. 

The error message I received is: 

```
Requested sites were configured for replication successfully.
Site replication error(s): 
'MakeBucketWithVersioning' on site s1 (44acd312-7293-4ca0-a3e1-8cc71fe1d05f): succeeded
'MakeBucketWithVersioning' on site z1 (d9a65ee4-e60e-4611-9386-4408ffe66dab): failed(s1->z1: MakeBucketWithVersioning: The request signature we calculated does not match the signature you provided. Check your key and signing method.)
```

I have tried changing the order of the site names (s1 and z1) in the command, but the error persists for site s1. 

Can someone please guide me on which parameters can affect the signature calculation? Are there any general things like DNS that could cause this issue?
Additionally, I should note that the distributed site uses HTTPS while the single node uses HTTP.
root  username ans password is same on both sides.

It's worth noting that other mc commands like status and info are working perfectly fine. 

Any help would be greatly appreciated.

**Comment by vadmeste** [comment, marked as answer]
@faryar76 's1' and 'z1' aliases need to have the correct endpoints and credentials of both sites, can you double check that ?

**Comment by poornas** [comment]
You can do a trace on z1 with `mc admin trace -a --status-code 403 -v z1` and check the signed headers used for Authorization - most likely that the host header has been modified if you are using a load balancer

**Comment by akkuman** [comment]
same error

**Comment by akkuman** [comment]
tldr: Just delete the data folder of the backup minio instance and add site again 

This problem is because the instance where the backup is located has been added before. If a site has been added, there will be a residual secretkey under/data/.minio.sys/config/iam/service-accounts/site-replicator-0/identity.json 

See https://github.com/minio/minio/blob/fd2c38fbefdb29649a262919fcbfc3bb518a0d43/cmd/site-replication.go#L384

https://github.com/minio/minio/blob/fd2c38fbefdb29649a262919fcbfc3bb518a0d43/cmd/site-replication.go#L469

The generated secretKey is then pushed to the peer (backup minio)via https://github.com/minio/minio/blob/fd2c38fbefdb29649a262919fcbfc3bb518a0d43/cmd/site-replication.go#L545 

If the peer has been previously configured for this site, the new and old secretkeys will not match, causing this problem 

So just delete the data folder of the backup minio instance and add site
